fedimint-client/src/transaction/sm.rs

```diff
@@ -146,37 +145,35 @@ impl State for TxSubmissionStatesSM {
 }
 
 impl TxSubmissionStates {
-    async fn trigger_created_rejected(tx: Transaction, context: DynGlobalClientContext) -> String {
+    async fn trigger_created_rejected(
+        transaction: Transaction,
+        context: DynGlobalClientContext,
+    ) -> String {
+        let mut backoff = custom_backoff(Duration::from_secs(2), Duration::from_secs(600), None);
+
         loop {
-            match context.api().submit_transaction(tx.clone()).await {
-                Ok(serde_outcome) => match serde_outcome.try_into_inner(context.decoders()) {
-                    Ok(outcome) => {
-                        if let TransactionSubmissionOutcome(Err(transaction_error)) = outcome {
-                            return transaction_error.to_string();
-                        }
-                    }
-                    Err(decode_error) => {
-                        warn!(target: LOG_CLIENT_NET_API, error = %decode_error, "Failed to decode SerdeModuleEncoding");
+            match context
+                .api()
+                .submit_transaction(transaction.clone())
+                .await
+                .try_into_inner(context.decoders())
+            {
+                Ok(outcome) => {
+                    if let TransactionSubmissionOutcome(Err(transaction_error)) = outcome {
+                        return transaction_error.to_string();
                     }
-                },
-                Err(error) => {
-                    error.report_if_important();
                 }
-            }
+                Err(decode_error) => {
+                    warn!(target: LOG_CLIENT_NET_API, error = %decode_error, "Failed to decode SerdeModuleEncoding");
+                }
+            };
 
-            sleep(RETRY_INTERVAL).await;
+            sleep(backoff.next().expect("Number of retries is has no limit")).await;
         }
     }
 
     async fn trigger_created_accepted(txid: TransactionId, context: DynGlobalClientContext) {
-        loop {
-            match context.api().await_transaction(txid).await {
-                Ok(..) => return,
-                Err(error) => error.report_if_important(),
```

Comment:
This reporting only when the error was not-expected and important was very helpful with debugging while avoiding spamming the logs on ever issue. We should add it to the new per-peer retry loop you've introduced, it would work quite well there I think.

Maybe we need to assign you more debugging weird crap issues and bugs so you get more sensitive about these. :P :D

Reply:
Added the report_if_important.